components/right-sidebar/SidebarSocialMedia.tsx

```diff
@@ -1,21 +1,65 @@
-import React from "react";
-import Link from "next/link";
-import SocialMediaLinks from "./SocialMediaLinks";
+import React, { useState } from "react";
 import Button from "../shared/ui/Button";
+import SocialMediaModal from "./SocialMediaModal";
+import { User, Social } from "@prisma/client";
+import { socialMediaIconList } from "@/lib/constants/socialMediaList";
+
+const SidebarSocialMedia = ({
+  user,
+}: {
+  user: User & { socialMedia?: Social[] };
+}) => {
+  const [open, setOpen] = useState(false);
 
-const SidebarSocialMedia = () => {
   return (
     <div className="flex flex-col gap-4">
-      <Link href="/profile/edit/update-socials">
-        <Button icon="plus" color="gray">
-          Update social link
-        </Button>
-      </Link>
-
-      <h3 className="paragraph-2-bold border-white-500 border-b pb-4">
+      <Button icon="plus" color="gray" onClick={() => setOpen(true)}>
+        Update social link
+      </Button>
+      <h3 className="paragraph-2-bold border-white-500 border-b pb-4 pt-4">
         Social Media Links
       </h3>
-      <SocialMediaLinks />
+
+      <div className="flex gap-2">
+        {user.socialMedia ? (
+          user.socialMedia.map((social) => {
+            const filtered = socialMediaIconList.filter(
+              (icon) => social.type === icon.type
+            );
+            const { icon: Icon } = filtered[0];
+
+            return (
+              <div className="flex items-center justify-center">
+                <Icon size={30} />
+                <a
+                  target="_blank"
+                  href={social.link}
+                  className="text-white-300 paragraph-2-regular"
+                >
+                  @{social.username}
+                </a>
+              </div>
+            );
+          })
```

Comment:
Struggled a little bit with mapping over the the socialMediaList  with the db list to match social media types, and t
hen to render the icon with the correct userName.  

Doing it this way is giving me styling issues...
<img width="255" alt="Screenshot 2024-03-28 at 10 18 47 AM" src="https://github.com/DDVVPP/GitNote/assets/47732125/136ad84f-a237-4f9b-9037-2f9e057cadc4">